### ISSUE TITLE
Add parameters for updated ami_analyze algorithm

### ISSUE THREAD

**Issue by stscijgbot-jp** [spacetelescope/jwst#5388]
_Issue [JP-1742](https://jira.stsci.edu/browse/JP-1742) was created on JIRA by [David Grumm](https://jira.stsci.edu/secure/ViewProfile.jspa?name=grumm):_

The additional parameters (although not required) in the configuration file improve the created Affine2d object in the updated algorithm.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1742?focusedCommentId=492056#comment-492056):_

The new parameters to be added should be documented here.

Also, there's no need to add them to the config file, just add them to the "spec" block in the step module, which defines all parameters for a given step. The config file should in fact not have any params listed in it. Normal operations should use defaults specified in the "spec".

**Comment by dmggh** [comment]
These are the new parameters that will be added:
set_psf_offset_find_rotation = string(default='0.0 0.0')   #  Values of the psf offset to use to create the model array
set_rotsearch_d = string(default='-3.0 3.1 1. ' ) # Set of values to use for the rotation search

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1742?focusedCommentId=500307#comment-500307):_

Also update the step docs to give info about the new params.

**Comment by jdavies-st** [comment]
It would be nice to have `configobj` take care off the validation as a list and float.  In the spec, that would look like:

```python
set_psf_offset_find_rotation = float_list(default=list(0, 0)) # Values of the psf offset to use to create the model array
set_rotsearch_d = float_list(default=list(-3, 3.1, 1)) # Set of values to use for the rotation search
```

This works fine for default values, but it seems `strun` will munge user inputs.  I.e. there's not a way (currently) to do:

```
$ strun ami_analyze.cfg input.fits --set_rotsearch_d="-3, 3.2, 1.5"
```

Bummer.  We should fix that.

**Comment by hbushouse** [comment]
Yes, it's always been the case that stpipe will not correctly parse any parameter that uses a list of values. That's why we have to always use ASN files as input to some steps, because it's impossible to specify a list of input files. I think it might work OK if you specify a list as a param value when calling from within python, but not from the command line. And you're right that we really should fix that. Are you volunteering?

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1742?focusedCommentId=500561#comment-500561):_

In the code we have:
```java
psf_offset_find_rotation = (0.0,0.0)
```
which result in a tuple, and:
```java
rotsearch_d = np.arange(-3, 3.1, 1)
```
which results in a numpy.ndarray object. So good luck figuring out how to define params and translate them into the equivalent objects.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1742?focusedCommentId=500606#comment-500606):_

I guess that for both of them the params could be setup to contain a list of 2 or 3 values, as [James Davies](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jdavies) suggested above, and then plug those individual values into the necessary places in the code lines above, e.g.
```java
psf_offset_find_rotation = (<param_name>[0], <param_name>[1])
rotsearch_d = np.arange(<param_name>[0]. <param_name>[1], <param_name>[2])```
Unless of course a real programmer like [James Davies](https://jira.stsci.edu/secure/ViewProfile.jspa?name=jdavies) has more elegant ways to do this.

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1742?focusedCommentId=500607#comment-500607):_

Or do the lowest-level workaround of defining them as strings, e.g. '-3 3.1 1', and then parsing the individual elements from the string.

**Comment by jdavies-st** [comment]
Yeah, it's a problem with our `argparse` implementation in `stpipe`.

https://github.com/spacetelescope/jwst/blob/5a8757199e3cdf01c75d2be2ed27cee4b6e2e1f2/jwst/stpipe/cmdline.py#L54-L98

A solution something like this

https://stackoverflow.com/questions/15753701/how-can-i-pass-a-list-as-a-command-line-argument-with-argparse

might work, and also solve (or something very similar would solve) the longstanding background step issue:

https://github.com/spacetelescope/jwst/issues/1720

**Comment by stscijgbot-jp** [comment]
_Comment by [Howard Bushouse](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bushouse) on [JIRA](https://jira.stsci.edu/browse/JP-1742?focusedCommentId=501211#comment-501211):_

Fixed by #5548
